airbyte/sources/base.py

```diff
@@ -582,6 +583,9 @@ def _execute(self, args: list[str]) -> Iterator[AirbyteMessage]:
                     message = AirbyteMessage.parse_raw(line)
                     if message.type is Type.RECORD:
                         self._processed_records += 1
+                        if self._last_stream_processed != message.record.stream:
+                            self._last_stream_processed = message.record.stream
+                            self._log_stream_read_start(self._last_stream_processed)
```

Comment:
With set logic, to avoid double-announcing:

```suggestion
                        if message.record.stream not in self._stream_names_observed:
                            self._stream_names_observed += message.record.stream
                            self._log_stream_read_start(message.record.stream)
```